src/servers/ZoneServer2016/zoneserver.ts

```diff
@@ -3245,6 +3309,96 @@ export class ZoneServer2016 extends ZoneServer2015 {
     }
     return inventory;
   }
+
+  clearMovementModifiers(client: Client) {
+    for (const a in client.character.timeouts) {
+      client.character.timeouts[a]._onTimeout();
+      delete client.character.timeouts[a];
+    }
+  }
+
+  applyMovementModifier(client: Client, modifier: number, type: string) {
+    this.multiplyMovementModifier(client, modifier);
+    switch (type) {
+      case "wellRested":
+        if (client.character.timeouts["wellRested"]) {
+          client.character.timeouts["wellRested"]._onTimeout();
```

Comment:
Are you sure it does what you want ? ._ontimeout() doesn't "destroy" the timer right ? so maybe clear it on this case "clearTimeout" instead of an if statement in the timer callback

Reply:
i tried to do it that way, somehow couldnt stop it from executing, thats why it checks if the timer still exists when it executes and returns the func if false

Reply:
I do it like that to execute it early, the movement modifier sucks, you cant just set it to 1 to be like before. if you multiplied the speed by 1.2, then you have to divide it by  1/1.2 to get it back to normal